### ISSUE TITLE
tf.keras.layers.LSTM and LSTMCell have different output formats

### ISSUE THREAD

**Issue by georgesterpu** [tensorflow/tensorflow#33145]
**System information**
- TensorFlow version (you are using): 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
The `tf.keras.layers.LSTM` and `tf.keras.layers.LSTMCell` classes currently have different output formats when using the `return_state=True` flag in `LSTM`. 

`LSTMCell` returns `[output, [h, c]]`
`LSTM` returns `[output, h, c]`

This makes it complicated to design a layer wrapper for both types of cells, since the wrapper's call method will have to take into account the cell type.
Are you okay with making `LSTM` adhere to the `LSTMCell` return format ?

**Will this change the current api? How?**
Yes. In `keras.layers.recurrent_v2.py` (Line 983):
```
    if self.return_state:
      #  return [output] + list(states)  # the current format
      return [output] + [states]  # the `LSTMCell` format
```

**Who will benefit with this feature?**
Me
**Any Other info.**
This possibly applies to other cells like GRU.

**Comment by qlzh727** [comment]
Thanks for reporting the issue. I can't easily update this to break the API contract. Could u provide more details about your wrapper? I am curious about it since LSTM layer and cell works differently (LSTM process the whole sequence, while the cell only process one time step).

**Comment by georgesterpu** [comment]
It is a simple dropout wrapper based on the polymorphic `tf.keras.layers.Dropout` class, designed for one of my tests. However, I don't think that I am going to use it from now on. Please feel free to close the issue if you think there are no other situations where this output uniformity could be useful.

**Comment by qlzh727** [comment]
I agree that the different format of output is bit confusing. The API contract predates my work for keras team, and changing it will be a somewhat expansive exercise.
